crates/sui-network/src/randomness/mod.rs

```diff
@@ -268,6 +269,12 @@ impl RandomnessEventLoop {
         // Aggregate any sigs received early from the new epoch.
         // (We can't call `maybe_aggregate_partial_signatures` directly while iterating,
         // because it takes `&mut self`, so we store in a Vec first.)
+        for ((epoch, round, peer_id), sig_bytes) in
+            std::mem::take(&mut self.future_epoch_partial_sigs)
+        {
+            // We can fully validate these now that we have current epoch DKG output.
```

Comment:
maybe add a debug_assert/warn that we get the expected epoch?

Reply:
We don't need that here because it's already checked extensively in receive_partial_signatures, if it still doesn't match it will be thrown away (if epoch is too old) or put back on the "future" queue (if too new)